admin/services/marking.service.js

```diff
@@ -43,23 +46,32 @@ markingService.mark = async function (completedCheck) {
     processedAt: moment.utc()
   }
 
+  const formData = JSON.parse(checkForm.formData)
```

Comment:
We should probably handle the `SyntaxError` that can be thrown from `JSON.parse`.  I would prefer it if the JSON was already parsed before it was passed to this `mark` function - this is usually done by using `checkForm.service` instead of `checkForm.data.service`

On line 44 of marking-service: `maxMarks` should be derived from the `checkForm` and not the supplied data.

Reply:
Hasn't been addressed.

Reply:
Just saw these comments will address them now